pfio/NetCDF4_FileFormatter.F90

```diff
@@ -246,17 +247,18 @@ subroutine open(this, file, mode, unusable, comm, info, rc)
          this%info = MPI_INFO_NULL
       end if
 
+      !$omp critical
       if (this%parallel) then
-         !$omp critical
          status = nf90_open(file, IOR(omode, NF90_MPIIO), comm=this%comm, info=this%info, ncid=this%ncid)
-         !$omp end critical
-         _VERIFY(status)
       else
-         !$omp critical
          status = nf90_open(file, IOR(omode, NF90_SHARE), this%ncid)
-         !$omp end critical
-         _VERIFY(status)
       end if
+      !$omp end critical
```

Comment:
A thought from @tclune as we looked at this: Should the `!$omp end critical` come *after* the write?

Reply:
Thanks for the suggestion. That make sense. I fixed it.